modules/core/js/src/main/scala/DynamoValue.scala

```diff
@@ -0,0 +1,261 @@
+      "S".colon(s.t.quotes)
+
+    def numbers(n: DynamoValue.Number) =
+      "N".colon(n.value.t.quotes)
+
+    def binaries(b: ByteVector) =
+      "B".colon(b.toBase64.t.quotes)
+
+    def bools(bool: Boolean) =
+      "BOOL".colon(Doc.str(bool))
+
+    def nuls =
+      "NULL".colon(Doc.str(true))
+
+    def lists(l: List[DynamoValue]) =
+      "L".colon(csv(l.map(_.toDoc.brackets)).squareBrackets)
+
+    def maps(m: Map[String, DynamoValue]) =
+      "M".colon {
+        csv {
+          m.toList.sortBy(_._1).map { case (k, v) => k.colon(v.toDoc.brackets) }
+        }.brackets
+      }
+
+    def stringSets(ss: NonEmptySet[String]) =
+      "SS".colon(csv(ss.value.map(_.t.quotes)).squareBrackets)
+
+    def numberSets(ns: NonEmptySet[DynamoValue.Number]) =
+      "NS".colon(csv(ns.value.map(_.value.t.quotes)).squareBrackets)
+
+    def binarySets(bs: NonEmptySet[ByteVector]) =
+      "BS".colon(csv(bs.value.map(_.toBase64.t.quotes)).squareBrackets)
+
+    this.fold(
+      strings,
+      numbers,
+      bools,
+      lists,
+      maps,
+      _ => nuls,
+      binaries,
+      binarySets,
+      numberSets,
+      stringSets
+    )
+  }.memoize
+
+  lazy val toDoc: Doc = lazyToDoc.value
+
+}
+object DynamoValue {
+
+  /** DynamoDb Number, which is represented as a string
+    */
+  case class Number(value: String)
+  object Number {
+    def of[A: Numeric](a: A): Number = {
+      // to avoid unused value warning
+      val _ = implicitly[Numeric[A]]
+      Number(a.toString)
+    }
+  }
+
+  val nul: DynamoValue = DynamoValue(AttributeValue.NULL)
+
+  def s(value: String): DynamoValue = DynamoValue(AttributeValue.S(value))
+
+  def bool(value: Boolean): DynamoValue = DynamoValue(
+    AttributeValue.BOOL(value)
+  )
+
+  def n(value: Number): DynamoValue = DynamoValue(AttributeValue.N(value.value))
+
+  def n[A: Numeric](value: A): DynamoValue =
+    n(Number.of(value))
+
+  def m(values: Map[String, DynamoValue]): DynamoValue =
+    DynamoValue(
+      AttributeValue.M(values.map { case (k, v) =>
+        (k, v.value)
+      }.toJSDictionary)
+    )
+
+  def m(values: (String, DynamoValue)*): DynamoValue =
+    m(values.toMap)
+
+  def l(values: List[DynamoValue]): DynamoValue = DynamoValue(
+    AttributeValue.L(values.map(_.value).toJSArray)
+  )
+
+  def l(values: DynamoValue*): DynamoValue =
+    l(values.toList)
+
+  def ss(values: NonEmptySet[String]): DynamoValue = DynamoValue(
+    AttributeValue.SS(values.value.toJSArray)
+  )
+
+  def ns(values: NonEmptySet[Number]): DynamoValue = DynamoValue(
+    AttributeValue.NS(values.value.map(_.value).toJSArray)
+  )
+
+  def bs(values: NonEmptySet[ByteVector]): DynamoValue = DynamoValue(
+    AttributeValue.BS(
+      values.value.map { b =>
+        new Uint8Array(b.toArray.map(_.toShort).toJSArray)
+      }.toJSArray
+    )
+  )
+
+  def b(value: ByteVector): DynamoValue = DynamoValue(
+    AttributeValue.B(new Uint8Array(value.toArray.map(_.toShort).toJSArray))
```

Comment:
```suggestion
    AttributeValue.B(value.toUint8Array)
```

Reply:
Hey, @armanbilge I have committed this (and the above). However, it requires the last version of scodec, which is compiled against scala 3.2.0 vs 3.0.0 that we were using here.

TASTy was complaining about that, so I have updated this build to use 3.2.0 as well; the question now is: Is it binary-compatible with the previous one? If not, as I assume, what should we do? bump up the minor version perhaps?

Thank you!